Description: Add support for bgr32 frame buffer device
  Plymouth can not recognize bgr32 frame buffer device.
  It used bgr32 format to flush argb32 device by default.
  So the color will be wrong.
Author: licheng <licheng@uniontech.com>

Origin: https://gerrit.uniontech.com/plugins/gitiles/base/plymouth/+/9a583b2560caeff5931cc531d940b831b4db7483
Bug: https://pms.uniontech.com/zentao/task-view-88626.html
Last-Update: 2022-05-18

--- plymouth-0.9.5.orig/src/plugins/renderers/frame-buffer/plugin.c
+++ plymouth-0.9.5/src/plugins/renderers/frame-buffer/plugin.c
@@ -512,6 +512,15 @@ query_device (ply_renderer_backend_t *ba
             backend->green_bit_position == 8 && backend->bits_for_green == 8 &&
             backend->blue_bit_position == 0 && backend->bits_for_blue == 8)
                 backend->flush_area = flush_area_to_xrgb32_device;
+        /*
+         * Add support for arm xbgr device.
+         */
+
+        else if (backend->bytes_per_pixel == 4 &&
+            backend->red_bit_position == 0 && backend->bits_for_red == 8 &&
+            backend->green_bit_position == 8 && backend->bits_for_green == 8 &&
+            backend->blue_bit_position == 16 && backend->bits_for_blue == 8)
+                backend->flush_area = flush_area_to_xrgb32_device;
         else
                 backend->flush_area = flush_area_to_any_device;
 
